saleor/tests/e2e/shop/utils/preparing_shop.py

```diff
@@ -1,44 +1,308 @@
+                        warehouse_ids=warehouse_ids,
+                        channel_ids=channel_ids,
+                    )
+                    shipping_zone_id = shipping_zone_data["id"]
+                    shipping_methods_for_zone = []
+
+                    for _ in range(zone_info.get("num_shipping_methods", 1)):
+                        shipping_method_data = create_shipping_method(
+                            e2e_staff_api_client, shipping_zone_id
+                        )
+                        shipping_method_id = shipping_method_data["id"]
+
+                        created_shipping_methods.append(shipping_method_data)
+                        shipping_methods_for_zone.append(shipping_method_id)
+                        created_shipping_methods_ids.append(shipping_method_id)
+
+                        create_shipping_method_channel_listing(
+                            e2e_staff_api_client,
+                            shipping_method_id,
+                            channel_id,
+                            price=shipping_price,
+                            minimum_order_price=minimum_order_price,
+                            maximum_order_price=maximum_order_price,
+                        )
+
+                        shipping_zone_data[
+                            "shippingMethods"
+                        ] = shipping_methods_for_zone
+                        created_shipping_zones.append(shipping_zone_data)
+
+            tax_config_data = get_tax_configurations(e2e_staff_api_client)
+            channel_tax_config = tax_config_data[0]["node"]
+            tax_config_id = channel_tax_config["id"]
+
+            update_tax_configuration(
+                e2e_staff_api_client,
+                tax_config_id,
+                tax_calculation_strategy=tax_calculation_strategy,
+                display_gross_prices=display_gross_prices,
+                prices_entered_with_tax=prices_entered_with_tax,
+            )
+
+            tax_rates = [
+                {
+                    "type": "country",
+                    "name": "Country Tax Class",
+                    "country_code": country,
+                    "rate": country_tax_rate,
+                }
+                if country_tax_rate is not None
+                else None,
+                {
+                    "type": "shipping",
+                    "name": "Shipping Tax Class",
+                    "country_code": shipping_country_code,
+                    "rate": shipping_country_tax_rate,
+                }
+                if shipping_country_tax_rate is not None
+                else None,
+                {
+                    "type": "billing",
+                    "name": "Billing Tax Class",
+                    "country_code": billing_country_code,
+                    "rate": billing_country_tax_rate,
+                }
+                if billing_country_tax_rate is not None
+                else None,
```

Comment:
Since we update country rates in each test maybe here we should only create tax classes for shipping, product, and product type?  

```suggestion
                {
                    "type": "shipping",
                    "name": "Shipping Tax Class",
                    "country_code": shipping_country_code,
                    "rate": shipping_country_tax_rate,
                }
                if shipping_country_tax_rate is not None
                else None,
```